### DISCUSSION TITLE
Specification of a new SAF System Functional Refinement VP

### DISCUSSION THREAD

**Discussion by maleckip** [GfSE/SAF-Specification#50] [Viewpoint]
There exist a System Functional Breakdown Viewpoint which represents the decomposition of system function(s) to adjunct system functions. 

**I have the impression, that there is no viewpoint representing the behavior realized by the adjunct system functions**. So, there is no way to see / validate, how the adjunct system functions contribute to the system function they are adjunct to. An activity diagram would be a good view I would expect. Having this kind of viewpoint, the System Functional Breakdown Viewpoint would be a kind of excerpt of it. One could name this (missing) viewpoint System Functional Behavior Viewpoint.

However, if the existing System Process Viewpoint has exactly this purpose, then the example provided doesn't express it well. It doesn't represent the behavior of "Detect and Report Fire", but the example for System Functional Breakdown Viewpoint displays the decomposition of  "Detect and Report Fire".

The adjunct system functions elicitation begins in a process with the proposed System Functional Behavior Viewpoint and has the System Functional Breakdown Viewpoint as a by-product / excerpt of it. Yes, one could define the adjunct system functions in a System Functional Breakdown Viewpoint as a brain dump, but their validation would remain pending.

Btw, I keep the term "adjunct" very unhandy / inadequate although it is defined by the UML.

**Comment by haarer** [comment]
Clarification needed.

**Comment by haarer** [comment]
Discussion:
System Process Viewpoints purpose is identification of system functions, not zooming into system functions (partial functions).
System Functional Breakdown Viewpoints purpose  shows the summary of partial functions per system function. (in a BDD)

Conclusion: New Viewpoint (AD), to allow deeper analysis of system functions into partial functions.

**Comment by haarer** [comment]
We have created a viewpoint, but it needs some work.
https://github.com/GfSE/SAF-Specification/blob/main/developing-saf/viewpoints/System-Functional-Refinement-Viewpoint.md
